external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -2937,12 +2938,88 @@ fn module_call_impl(
     (call, return_)
 }
 
+/// If the constant that we are referencing has an `error` attribute, we need to change the type of
+/// the constant to a u64 since this will be compiled into a u64 error code.
+fn annotated_error_const(context: &mut Context, e: &mut T::Exp) {
+    let sp!(
+        const_loc,
+        T::UnannotatedExp_::Constant(module_ident, constant_name)
+    ) = &mut e.exp
+    else {
+        return;
```

Comment:
Just realized. What happens if you do something like `assert!(false, b"error")`?

Reply:
Nothing should change from the behavior today -- you'd get a type error that `b"error"` isn't a u64.